pandas/core/generic.py

```diff
@@ -8876,15 +8868,12 @@ def _align_frame(
             right = right.fillna(method=method, axis=fill_axis, limit=limit)
 
         # if DatetimeIndex have different tz, convert to UTC
-        if is_datetime64tz_dtype(left.index.dtype):
-            if left.index.tz != right.index.tz:
-                if join_index is not None:
-                    # GH#33671 ensure we don't change the index on
-                    #  our original Series (NB: by default deep=False)
-                    left = left.copy()
-                    right = right.copy()
-                    left.index = join_index
-                    right.index = join_index
```

Comment:
IIRC one of these two usages doesnt have test coverage

Reply:
true, but that's not affected by this PR - one of the two calls to `_set_join_index` will still be uncovered (as before) and the other one won't